File: 4multilevel_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;

// Parent class
class Vehicle {
public:
    void start() {
        cout << "Vehicle started" << endl;
    }
};

// Parent class inheriting from Vehicle
class Car : public Vehicle {
public:
    void drive() {
        cout << "Car is being driven" << endl;
    }
};

// Child class inheriting from Car
class ElectricCar : public Car {
public:
    void charge() {
        cout << "Electric car is charging" << endl;
    }
};


int main(){
    
    
    ElectricCar my_electric_car;
    my_electric_car.start(); 
    my_electric_car.drive();  
    my_electric_car.charge();
    return 0;
}

File: 4multilevel_inheritance.py
# Parent class
class Vehicle:
    def start(self):
        print("Vehicle started")

# Parent class inheriting from Vehicle
class Car(Vehicle):
    def drive(self):
        print("Car is being driven")

#  Child class inheriting from Car
class ElectricCar(Car):
    def charge(self):
        print("Electric car is charging")

my_electirc_car = ElectricCar()
my_electirc_car.start()
my_electirc_car.drive()
my_electirc_car.charge()